Cleanup reference build artifacts

the-grue opened the pull request:
This change moves all previously built objects out of the sources/src/*/ directories into /sources/src/*/ref.

For historic purposes as well as validation (via checksum, etc) of new builds against previous builds it is better to keep these artifacts around.  This solution gets them out of the way from the build tree, but keeps them close at hand for comparision.

This solution is open for consideration as it changes the layout of the source tree, but I feel it is a good solution.

andrewbird commented:
Firstly the goal here is fine: to move the files that would be changed by a build to a reference subdirectory. I have a few observations:
- You should probably work in a git branch and rebase it off the current master before raising the PR, it will cut down on the merge commits and you'll get to fix any conflicts with latest code before submission.
- It seems that there are some source/build files being copied `EXE2BIN.DAT` to the reference directory.
- There are source changes in the unterm directory, why? If necessary, they should be a separate commit from the renaming commit.

A followup commit might look at identifying duplicates within the non reference directories, but I don't want to distract you from your quest for #2

roelandjansen commented:
I agree with @andrewbird -- atm it has no conflicts with the base branch but the two other points are valid.

the-grue commented:
1. Yeah, sorry, still trying to figure out how some of this git stuff works.  I had to pull the latest changes from Roeland's repo (b6c8400), made my changes (7bbba32), and resync my on-disk code with b6c8400 before it would let me push 7bbba32 (bb786cb).  Any pointers how to only present 7bbba32?  The other two only reflect me trying to snyc my repo with Roeland's.

2. My bad on that, I moved them to the ref directory and then pulled them back down when I realized they were used by debug and forgot to remove them from ref afterwards.

3. The unterm directory only had a zip file in it.  I extracted the zip file and pushed the zip up to ref.  This code actually builds.  I believe the zip files in the other directories had been extracted in the same fashion on initial upload, so this makes things consistent.

I'll try to redo my pull request in a proper fashion.

the-grue commented:
Please see #37
